.github/workflows/test-suite.yml

```diff
@@ -2,24 +2,47 @@ name: 'Test Suite'
 
 on:
   workflow_dispatch:
+    inputs:
+      terraform_matrix:
+        description: 'Terraform version matrix (JSON array)'
+        default: '["1.6.x", "1.5.x", "1.4.x", "1.3.x", "1.2.x"]'
+      provider_matrix:
+        description: 'MongoDB Atlas Provider version matrix (JSON array)'
+        default: '["", "1.12.3"]' # "" for latest version
```

Comment:
should we still have 1.12.3 hardcoded here? I think what we want ultimately is that our latest version is compatible... right @Zuhairahmed ?

Reply:
@marcosuma Zuhair told me to check our latest 2 minor patch versions, "" is the latest one (currently 1.13.1).
for example once we release 1.14.0,   we'll have to change this to:  "", "1.13.1", where "" will be 1.14.0

Reply:
I don't understand why 😅 and also this is something we'll have to manually update too often, every time we make a release (yet one more thing to do manually). Unless there is a strong reason (let's wait for Zuhair) that I am missing let's please remove it.

Reply:
is it just me or I am really not seeing the value in this?

Once you've gone over 1.12.3, what is the point in keep testing the second last version? That will never change because we will never touch it.

To me it's more the extra manual effort we are introducing in having to edit the last version every time than the benefit of this

@lantoli @Zuhairahmed

Reply:
would consider the HashiCorp Terraform testing scenarios where: 
1) acceptance tests fail on Terraform Atlas Provider v1.13.1 and pass on v1.12.3   
2) acceptance tests fail on both Terraform Atlas Provider v1.13.1 and v1.12.3

if we only tested most recent version all we would know for certain is that tests have failed on v1.13.1 and we need to investigate why this occurred which can be from multiple sources. however by running testing on most 2 recent recent version of Terraform Atlas Provider we help accelerate triage of finding a bug by narrowing scope of issue. if scenario 1 one occurs, then we know that error is something that we introduced with most recent changes catching breaking changes ahead of customers. if it's scenario 2, then we know that issue most likely has to do with change made on HashiCorp Terraform side (for example they released bug) so we can then escalate to HashiCorp thereby saving time with bug resolution.  

Does this help? Agree again that we should be automating/avoiding hardcodes as much of this as possible.

Reply:
cc. @lantoli 

Thanks @Zuhairahmed for clarifying it, I still believe it does not add any valuable information.
First of all, the main blocker for me is the hard-coded value: in a team that is giving it all to fully automate things, I am really against hardcoding things as it moves to the opposite direction (this code means every time we create new release, we have to update this value).
Even if we wanted to keep it, I don't understand why we chose 1.12.3, since eventually we are at 1.14.0 and we should have had ("latest", "1.13.1"). So maybe @lantoli @Zuhairahmed can you also clarify that? Surely I am missing something here.


Said that, I still think it's not a good idea because we are not considering here that tests have a cost for the company since we are constantly calling the backend: they are not for free, we are paying for that.


This is my rationale. Let's give names to things:
- latest: is the latest version released on TF for our provider (e.g. 1.14.0)
- second-latest: self-explanatory (e.g. 1.13.1)
- master: this is whatever is on the branch master, which hasn't been released yet
- let's also remind that **there is no back-porting**, neither for TF version nor for our provider.

At time _t0_ we released _"second-latest"_: as per our process we've released it by checking that all tests are passing, otherwise we don't even release. This means it is not possible for a release to be released with failing tests, otherwise we have made a mistake that we need to fix right away.

Let's say at time _t1_ we have released _"latest"_: same answer as before.

Later, at time _t2_ we have made other changes on _"master"_ and one test starts failing. The test that fails can be only because:
1. we introduced a bug right now on master
2. we started using the newest feature from TF that breaks previous compatibility. 
3. there is a server error in the **Atlas backend** 
4. there is a bug introduced in the **latest Atlas APIs**
5. it **cannot be** because an older terraform version became buggy
6. it **cannot be** because an older provider version became buggy

With this new code we just merged we are constantly testing also _"second-latest"_: can you please clarify to me what kind of value does it bring in this investigation? What is it that adds to the RCA that logs already don't have to identify what is the problem? Keep in mind that:
- if either 1. or 2. is true that is trivial, we've just done an action that generated an error (it's a matter or reverting a commit to verify that)
- if 3. is true then all versions will fail. 
- if 4. is true only the latest version will fail, but we already know from the logs similarly to what happened in CLOUDP-218507 and CLOUDP-212755